azavea/ashlar-blueprint
# Build out a backend schema editor

## Overview

This PR pulls out the Ashlar-specific components of the DRIVER schema editor and incorporates them as a backend to the Ashlar reference implementation.

### Checklist


### Demo

Navigate to `localhost:9000`, log in, and view the editor up and running:

![ashlar-editor](https://user-images.githubusercontent.com/14170650/41560019-d6aca7c8-7313-11e8-94e1-d0b5c76536f3.png)

### Notes

- The diff here is regrettably large, since I'm essentially porting another app into this repo. You can safely ignore changes in the following repos, since they're almost exclusively subtractive from what exists in DRIVER:
    - `ashlar/ashlar_auth` (authentication component of Ashlar, forked from `driver/app/driver_auth`)
    - `schema_editor`
- It strikes me that the editor should probably live in its own repo in the same way that Ashlar lives in its own repo, separate from this implementation. Unfortunately there's some hard-wiring between both services that is leftover from DRIVER; once the first draft of the reference implementation is reasonably complete, it'd be nice to take a step back and fully abstract these components out into their own repos.

## Testing Instructions

Run the tests:

- Update builds and dependencies with `./scripts/update` 
- Run `./scripts/test` to run all tests

Run the development server and poke around to make sure everything looks good:

- Update builds and dependencies with `./scripts/update`
- Run the development server with `./scripts/server`
- Navigate to the schema editor at the origin `localhost:9000`, login with the username `admin` and the password `admin`, and browse around

Closes https://github.com/azavea/ashlar-2018-fellowship/issues/5.
